src/main/scala/scorex/core/ModifiersCache.scala

```diff
@@ -0,0 +1,139 @@
+package scorex.core
+
+
+import scorex.core.consensus.HistoryReader
+import scorex.core.utils.ScorexLogging
+import scorex.core.validation.RecoverableModifierError
+
+import scala.collection.mutable
+import scala.util.{Failure, Success}
+
+/**
+  * A cache which is storing persistent modifiers not applied to history yet.
+  * @tparam PMOD - type of a persistent node view modifier (or a family of modifiers).
+  */
+trait ModifiersCache[PMOD <: PersistentNodeViewModifier, H <: HistoryReader[PMOD, _]] {
+  require(maxSize >= 1)
+
+
+  protected val cache = mutable.Map[K, V]()
```

Comment:
I remember we had a convension, that class should define type parameters first, then variables and functions at the end. In this trait order looks random

Reply:
fixed